### PR TITLE
Memory leak checker API.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: runtime/src/main/kotlin/kotlin/native/Platform.kt [hunk 1]
```diff
@@ -81,6 +81,14 @@ public object Platform {
     public val isDebugBinary: Boolean
         get() = Platform_isDebugBinary()
 
+    /**
+     * If the memory leak checker is activated, by default true in debug mode, false in release.
+     * When memory leak checker is activated, and leak detected on process termination process
+     * is terminated with non-zero exit code.
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/kotlin/kotlin/native/Platform.kt]: Is it?

File: runtime/src/main/kotlin/kotlin/native/Platform.kt [hunk 2]
```diff
@@ -81,6 +81,14 @@ public object Platform {
...
+     */
+    public var hasMemoryLeakChecker: Boolean
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/kotlin/kotlin/native/Platform.kt]: The name doesn't correspond to the behaviour.
Maybe `isMemoryLeakCheckerEnabled`?

File: runtime/src/main/kotlin/kotlin/native/Platform.kt (hunk 1 above)
Comment (olonho) [reply runtime/src/main/kotlin/kotlin/native/Platform.kt]: Oh, forgot to push update doc.

Comment (SvyatoslavScherbina) [reply runtime/src/main/kotlin/kotlin/native/Platform.kt]: Error message without termination is somewhat indiscoverable (including leaks in Kotlin/Native tests). Why not terminate?

Comment (olonho) [reply runtime/src/main/kotlin/kotlin/native/Platform.kt]: Well, OK, will update message as well.

File: runtime/src/main/kotlin/kotlin/native/Platform.kt [hunk 3]
```diff
@@ -82,11 +82,11 @@ public object Platform {
         get() = Platform_isDebugBinary()
 
     /**
-     * If the memory leak checker is activated, by default true in debug mode, false in release.
-     * When memory leak checker is activated, and leak detected on process termination process
-     * is terminated with non-zero exit code.
+     * If the memory leak checker is activated, by default `true` in debug mode, `false` in release.
+     * When memory leak checker is activated, and leak is detected during last Kotlin/Native runtime
+     * deinitialization process - error message with leak information is printed.
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/kotlin/kotlin/native/Platform.kt]: > during last Kotlin/Native runtime deinitialization process

This is unclear. Maybe align the terms with workers?

File: runtime/src/main/kotlin/kotlin/native/Platform.kt (hunk 2 above)
Comment (olonho) [reply runtime/src/main/kotlin/kotlin/native/Platform.kt]: Renamed.

File: runtime/src/main/kotlin/kotlin/native/Platform.kt (hunk 3 above)
Comment (olonho) [reply runtime/src/main/kotlin/kotlin/native/Platform.kt]: What would you suggest?

Comment (SvyatoslavScherbina) [reply runtime/src/main/kotlin/kotlin/native/Platform.kt]: Workers seem to use "Kotlin context": https://github.com/JetBrains/kotlin-native/blob/a98e5c799cd7612b014b6bc926c899922eea2453/runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt#L42

I suggest using the same phrase here, maybe described more verbosely (like "workers and platform threads ran Kotlin code"). Or maybe keep only the description, without "Kotlin context".

**Pull request merged**